cpp/src/arrow/util/basic_decimal.h

```diff
@@ -166,6 +167,317 @@ class ARROW_EXPORT GenericBasicDecimal {
   }
 };
 
+template <typename BaseType>
```

Comment:
`BaseType` sounds like a base class, perhaps call it `DigitType` or simply `Digit`.

Reply:
renamed